src/ps/handlers/interface.ts

```diff
@@ -26,11 +26,12 @@ export function interfaceHandler(message: PSMessage) {
 			if (!game) return; // Don't put any errors here! People should be able to close games that don't exist, like ones that ended
 
 			const user = message.author.id;
-			if (game.hasPlayer(user)) {
+			const player = Object.values(game.players).find(p => p.id === user);
+			if (game.hasPlayer(user) && player && !player.out) {
 				message.reply(game.$T('GAME.CANNOT_LEAVE', { prefix, game: game.meta.id }));
 				return game.update(user);
 			}
-			if (game.spectators.includes(user)) {
+			if (game.spectators.includes(user) || (player && player.out)) {
```

Comment:
Oh oops

Issue here is that a player who's `out` won't be in spectators either. We might need to do something like:
* Players are only sent HTML if `player.out` is not `true`
* Players that should keep spectating even upon being `out` (eg: Chain Reaction) should be moved to `game.spectators` instead (and stuff like `rejoin` will have to be rechecked once to make sure `player.out === true && spectators.includes(player)` conditions are handled

Reply:
Should be able to remove the line here now

Also need to disable auto-updates for players that are out

Reply:
![image](https://github.com/user-attachments/assets/cff0f221-67c8-49d7-89fc-7492a2a2478b)
(The second case is handled correctly but not the first, so if anything happens that tries to fire a rejoin, they'll be rendered as a full player)